src/agent/transform-headers-agent.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-underscore-dangle */
+const http = require('http');
+const WrappedAgent = require('./wrapped-agent');
+
+const { _storeHeader } = http.OutgoingMessage.prototype;
+
+class TransformHeadersAgent extends WrappedAgent {
+    transformRequest(request) {
```

Comment:
This has been rewritten from [this](https://github.com/nodejs/node/blob/533cafcf7e3ab72e98a2478bc69aedfdf06d3a5e/lib/_http_outgoing.js#L442-L479). Also removed parts related to server-side, as this is a client.